src/main/java/homework/seven/ArrayDataException.java

```diff
@@ -0,0 +1,19 @@
+package homework.seven;
+
+public class ArrayDataException extends Exception {
+    public ArrayDataException() {
```

Comment:
Do you really use all of these constructors? If not - not needed code should be dropped